archunit/src/test/java/com/tngtech/archunit/core/domain/JavaPackageTest.java

```diff
@@ -241,77 +241,151 @@ public void visits_packages() {
 
-        assertThatDependencies(examplePackage.getPackage("unrelated").getClassDependenciesToSelf())
+        assertThat(examplePackage.getPackage("third").getPackageDependenciesFromThisPackage())
                 .isEmpty();
+
+        assertThatPackages(examplePackage.getPackage("unrelated").getPackageDependenciesFromThisPackage())
+                .containOnlyNames("java.lang");
     }
 
     @Test
-    public void has_package_dependencies_to_other_packages() {
+    public void has_package_dependencies_from_this_package_and_subpackages() {
         JavaPackage examplePackage = importPackage("packageexamples");
 
-        assertThat(examplePackage.getPackage("second").getPackageDependenciesFromSelf())
+        assertThat(examplePackage.getPackage("second").getPackageDependenciesFromThisPackageAndSubpackages())
                 .containsOnly(
                         getRoot(examplePackage).getPackage("java.lang"),
                         examplePackage.getPackage("first"),
                         examplePackage.getPackage("third.sub"));
 
-        assertThat(examplePackage.getPackage("third").getPackageDependenciesFromSelf())
+        assertThat(examplePackage.getPackage("third").getPackageDependenciesFromThisPackageAndSubpackages())
                 .containsOnly(examplePackage.getPackage("first"));
 
-        assertThatPackages(examplePackage.getPackage("unrelated").getPackageDependenciesFromSelf())
+        assertThatPackages(examplePackage.getPackage("unrelated").getPackageDependenciesFromThisPackageAndSubpackages())
                 .containOnlyNames("java.lang");
     }
 
     @Test
-    public void has_package_dependencies_from_other_packages() {
+    public void has_package_dependencies_to_this_package() {
         JavaPackage examplePackage = importPackage("packageexamples");
 
-        assertThat(examplePackage.getPackage("first").getPackageDependenciesToSelf())
+        assertThat(examplePackage.getPackage("first").getPackageDependenciesToThisPackage())
                 .containsOnly(
                         examplePackage.getPackage("second"),
                         examplePackage.getPackage("second.sub"),
                         examplePackage.getPackage("third.sub"));
 
-        assertThat(examplePackage.getPackage("third").getPackageDependenciesToSelf())
-                .containsOnly(examplePackage.getPackage("second.sub"));
+        assertThat(examplePackage.getPackage("second").getPackageDependenciesToThisPackage())
+                .doesNotContain(examplePackage.getPackage("second"));
+
+        assertThat(examplePackage.getPackage("third").getPackageDependenciesToThisPackage())
+                .isEmpty();
 
-        assertThat(examplePackage.getPackage("second").getPackageDependenciesToSelf())
+        assertThat(examplePackage.getPackage("unrelated").getPackageDependenciesToThisPackage())
+                .isEmpty();
+    }
+
+    @Test
+    public void has_package_dependencies_to_this_package_and_subpackages() {
```

Comment:
Except for `.getPackage("third")` this case seems to be identical with the one before. Should we add another class, or move a class within `"first"` to a subpackage to make the first assertion more clearly different?

Reply:
Now that you have fixed the bug mentioned here: https://github.com/TNG/ArchUnit/pull/922/files#r956754244 the assertion for `second` is also different, so probably it's okay as-is.